src/main/java/teammates/ui/controller/InstructorCourseEditSaveAction.java

```diff
@@ -14,6 +15,11 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
         Assumption.assertPostParamNotNull(Const.ParamsNames.COURSE_ID, courseId);
 
+        // Restore previous attributes of the course
+        CourseAttributes course = logic.getCourse(courseId);
```

Comment:
The course may not exist. You need to handle that.

Reply:
Actually, I think you don't need to retrieve the course and see which fields are changed. Just have a log message saying the course is updated and list down the new values of all fields, similar to how it's done for feedback session:
https://github.com/TEAMMATES/teammates/blob/cd9a5365cfed580bae5d1bf15f2994a921e31f03/src/main/java/teammates/ui/controller/InstructorFeedbackEditSaveAction.java#L36-L50

Reply:
This is my first PR and I am fairly new to the system so please bear with me if I ask any stupid questions.

I noticed that ```logic.getCourse(courseId)``` can return null if the course doesn't exist but I was wondering under what circumstances can the user edit a course that doesn't exist in DB.

Also, the original [issue](https://github.com/TEAMMATES/teammates/issues/8695) mentioned that the log message should only display old and new fields of attributes that have changed that's why I did it this way. Should I follow the pattern for feedback session instead ?

Reply:
> This is my first PR and I am fairly new to the system so please bear with me if I ask any stupid questions.
No worries. We are all here to learn.

> I noticed that logic.getCourse(courseId) can return null if the course doesn't exist but I was wondering under what circumstances can the user edit a course that doesn't exist in DB.
`courseId` is a parameter that is totally under user's control, so a malicious user may wants to crash the program and send an invalid Id.

> Also, the original issue mentioned that the log message should only display old and new values of fields that have changed that's why I did it this way. Should I follow the pattern for feedback session instead ?
The expected output described in the issue is the ideal behaviour. However, with the current structure of the app, it needs to be done like how you did. This wastes database queries, which is a precious resource, while the extra information that is displayed is not important. So I suggest the simpler method of just displaying all the parameters regardless of whether they are different from the previous values or not.

Reply:
Ok I see. Sure I'll make the changes accordingly. Thanks for the clarifications :)